lib/Commands/NinjaBuildCommand.cpp

```diff
@@ -1681,7 +1681,7 @@ std::unique_ptr<core::Rule> NinjaBuildEngineDelegate::lookupRule(const core::Key
   // FIXME: This is frequently a redundant lookup, given that the caller might
   // well have had the Node* available. This is something that would be nice
   // to avoid when we support generic key types.
-  ninja::Node* node = context->manifest->findOrCreateNode(workingDirectory, key);
+  ninja::Node* node = context->manifest->findOrCreateNode(workingDirectory, key.str());
```

Comment:
Should this method just take the key?

Reply:
In this case it won't get us much.  The code takes the StringRef that this is implicitly converted to and immediately moves it into a SmallString.  It also passes the StringRef as is to a Node().   Changing that all to use KeyTypes is probably a good idea, but out of scope here.